pkg/kubectl/cmd/util/printing.go

```diff
@@ -126,6 +127,15 @@ func PrinterForCommand(cmd *cobra.Command, outputOpts *printers.OutputOptions, m
 		return nil, err
 	}
 
+	// if printer.isGeneric() not true, it must be a HumanReadablePrinter
+	// we execute AddHandlers() here before maybeWrapSortingPrinter so that we don't
+	// need to use delegatePrinter convert again outsider when it's a SortingPrinter
```

Comment:
outsider means ring2Factory.PrinterForMapping this line:https://github.com/kubernetes/kubernetes/pull/48788/files#diff-a9b5082d3df09126fbcd118712148907R113,